src/app/(account)/settings/tabs/AccountSettings.tsx

```diff
@@ -1,147 +1,202 @@
-                } else {
-                    setMembershipStatus('Payment Required');
-                }
-            } catch (error) {
-                console.error('Error verifying membership payment:', error);
-            }
-        };
+        // Avoid creating duplicate email addresses. Find email address (unverified) first.
+        let newEmailInst = user?.emailAddresses.find(
+            ({ emailAddress }) => emailAddress === data.email
+        );
+        if (!newEmailInst) {
+            newEmailInst = await user?.createEmailAddress({ email: data.email });
+        }
+        await newEmailInst?.prepareVerification({ strategy: 'email_code' });
 
-        verifyMembershipPayment();
+        setIsLoading(false);
+        setIsCodeSent(true);
     });
 
-    return (
-        <div className="flex flex-col gap-8">
-            {membershipStatus === 'Payment Required' && (
-                <div>
-                    <h2 className="text-2xl font-bold">
-                        Membership Status: <span className="text-orange">Payment Required</span>
-                    </h2>
+    const codeForm = useForm<z.infer<typeof codeSchema>>({
+        defaultValues: { code: '' },
+        resolver: zodResolver(codeSchema),
+    });
+    const updateEmail = useSWRMutation('member/email', fetcher.patch.mutate, {});
+    const handleUpdateEmail = codeForm.handleSubmit(async ({ code }) => {
+        setIsLoading(true);
+
+        const newEmailInst = user?.emailAddresses.find(
+            ({ emailAddress }) => emailAddress === newEmail
+        );
+        try {
+            await newEmailInst?.attemptVerification({ code });
+        } catch {
+            codeForm.setError('code', {
+                message: 'Incorrect Code. Please enter the code from your new email.',
+            });
+            setIsLoading(false);
+            return;
+        }
+        await updateEmail.trigger({ code, email: newEmail });
 
-                    <div className="mb-6 border-b-2 border-black" />
-                    <p>
-                        Finalise your membership by{' '}
-                        <span
-                            className="cursor-pointer font-bold text-purple"
-                            onClick={handleGoToMembership}
+        setIsLoading(false);
+        window.location.reload();
+    });
+
+    return (
+        <div className="space-y-4">
+            <h2 className="text-2xl font-bold">Change Email</h2>
+            <div className="mb-2 border-b-2 border-black" />
+            <div>
+                Current Email: <b>{currentEmail}</b>
+            </div>
+            {!isCodeSent && (
+                <form onSubmit={handleSendCode}>
+                    <ControlledField control={emailForm.control} name="email" label="New Email" />
+                    <Button colour="orange" loading={isLoading}>
+                        Send Code
+                    </Button>
+                </form>
+            )}
```

Comment:
Should probably have a sentence here saying "A verification email containing a code will be sent for you to confirm your new email address"